website/docs/scanner-plugins.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Scanner Plugins
+---
+
+# Motivation
+
+By default, `copa` uses [Trivy](https://github.com/aquasecurity/trivy) to scan container images for vulnerabilities. However, we understand that different organizations have different requirements and may want to use different vulnerability scanners.
+
+Starting with v0.5.0 and later, `copa` offers extensibility to support different vulnerability scanners. Plugin architecture allows users to use the vulnerability scanner of their choice to patch container images without having to modify `copa`'s core codebase.
+
+# Usage
+
+Scanner plugin binaries must be in `$PATH` and should be prefixed with `copa-`. Copa will automatically detect and use the scanner plugin if it is in `$PATH`.
+
+For example, if you have a scanner plugin binary called `copa-foo` in `$PATH`, you can run `copa` with the following command:
+
+```bash
+copa patch --scanner foo --image $IMAGE ...
+```
+
+# Scanner Plugins from the Community
+
+If you have built a scanner plugin and would like to add it to this list, please submit a PR to update this section with your plugin.
+
+:::tip
+
+If you have any issues with a specific plugin, please open an issue in the applicable plugin's repository.
+
+:::
+
+- Grype: https://github.com/anubhav06/copa-grype
+
+# Writing a Scanner Plugin
+
+Please see instructions at [Scanner Plugin Template](https://github.com/project-copacetic/scanner-plugin-template) for a template to get started with writing a scanner plugin.
+
+# Scanner Plugin Interface
+
+Scanner plugins must implement the following interface:
+
+## v1alpha1
+
+```golang
+type UpdateManifest struct {
+    // API version of the interface (e.g. v1alpha1)
+    APIVersion string         `json:"apiVersion"`
+    // Metadata contains information about the OS and config
+    Metadata   Metadata       `json:"metadata"`
+    // Updates is a list of UpdatePackage that contains information about the package updates
+    Updates    UpdatePackages `json:"updates"`
+}
+
+// UpdatePackages is a list of UpdatePackage
+type UpdatePackages []UpdatePackage
+
+// Metadata contains information about the OS and config
+type Metadata struct {
+    OS     OS     `json:"os"`
```

Comment:
does copa validate against a supported list of OS and Arch?

Reply:
or do they non-supported ones just get skipped?

Reply:
for OS, yes it'll validate and fail fast if it's not supported
for Arch, it's used for pulling the arch-specific image and vex output